### PR TITLE
Update Ethereum tests spec to v10.4

### PR DESCRIPTION
**Pull request by Stefan-Ethernal**
# Description

Updated test cases suite to v10.4. This version includes london hardfork test cases and does not inlude the merge one and all further updates. Currently, london hardfork test cases are ignored since there are some differences in gas consumption. 

Also, TestEVM is not needed anymore since those test cases were moved to the regular state tests. Eventually, if there is any difference in the tx execution, the block hash is going to be different and a test will fail. 

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [ ] I have added the relevant labels
- [ ] I have updated the official documentation
- [ ] I have added sufficient documentation in code

## Testing

- [ ] I have tested this code with the official test suite
- [ ] I have tested this code manually

### CODE REVIEW

**Review by vcastellm — APPROVED** [review]
LGTM

File: tests/state_test.go [hunk 1]
```diff
@@ -15,40 +15,57 @@ import (
...
-	msg, err := c.Transaction.At(p.Indexes)
+	var baseFee *big.Int
+
+	if config.IsLondon(0) {
+		if c.Env.BaseFee != "" {
+			baseFee = stringToBigIntT(t, c.Env.BaseFee)
+		} else {
+			// Retesteth uses `0x10` for genesis baseFee. Therefore, it defaults to
+			// parent - 2 : 0xa as the basefee for 'this' context.
+			baseFee = big.NewInt(0x0a)
```
Comment (igorcrevar) [inline tests/state_test.go]: maybe put this 0x0a value inside some const?

**Review by igorcrevar — APPROVED** [review]
Approved, just a small neat-pick

File: tests/state_test.go (hunk 1 above)
Comment (begmaroman) [reply tests/state_test.go]: Done

**Pull request merged**